src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/SummaryListWriter.java

```diff
@@ -256,7 +256,7 @@ protected Content getSummaryLink(Element e) {
             case ENUM_CONSTANT -> new EnumConstantWriterImpl(this);
             case RECORD_COMPONENT ->
                 throw new AssertionError("Record components are not supported by SummaryListWriter!");
-            default -> new AnnotationTypeOptionalMemberWriterImpl(this, null);
+            default -> new AnnotationTypeMemberWriterImpl(this);
```

Comment:
1. I see it was always this way, but maybe it would be better to have an explicit `case` label, and have `default` throw some sort of `not handled` exception, or else rely on javac doing a completeness check.
2. I'm surprised there isn't any parameter to the new constructor to indicate whether optional or required members are required.

Reply:
This one surprised me: Annotation elements do not have an `ElementKind` constant of their own, they use `ElementKind.METHOD`. This means the code above was never executed. It is also the reason annotation elements are linked as methods, including the empty parentheses. I'm not sure whether this is an oversight, a bug, or there is some intention behind it.

As for the code above I think it can be replaced with throwing a UnsupportedOperationException.